### PR TITLE
Use module hashes in blueprints

### PR DESCRIPTION
**Pull request by folex**

### CODE REVIEW

File: aquamarine/Cargo.toml [hunk 1]
```diff
@@ -10,7 +10,7 @@ particle-protocol = { path = "../particle-protocol"}
 fluence-libp2p = { path = "../crates/libp2p" }
 config-utils = { path = "../crates/config-utils" }
 host-closure = { path = "../crates/host-closure" }
-aquamarine-vm = "0.1.29"
+aquamarine-vm = { git = "https://github.com/fluencelabs/fce", branch = "file_name" }
```
Comment (michaelvoronov) [inline aquamarine/Cargo.toml:13]: publish and change?

File: particle-modules/src/lib.rs [hunk 2]
```diff
@@ -14,9 +14,10 @@
  * limitations under the License.
  */
 
+#![feature(try_blocks)]
 #![warn(rust_2018_idioms)]
 #![deny(
-    dead_code,
+    // dead_code,
```
Comment (michaelvoronov) [inline particle-modules/src/lib.rs:20]: Is it intended?

File: particle-modules/src/modules.rs [hunk 3]
```diff
@@ -14,109 +14,213 @@
...
-                        .map_err(|err| {
-                            log::warn!("failed to deserialize blueprint {}: {:#?}", fname, err)
-                        })
-                        .ok()?;
-
-                    // Convert to json
-                    serde_json::to_value(blueprint).ok()
+                .filter_map(|path| {
+                    let hash = extract_module_name(&path)?;
+                    let result: eyre::Result<_> = try {
```
Comment (michaelvoronov) [inline particle-modules/src/modules.rs:127]: What is `eyre`?)

File: particle-node/Cargo.toml [hunk 4]
```diff
@@ -53,8 +53,9 @@ fstrings = "0.2.3"
 particle-providers = { path = "../particle-providers"}
 test-utils = { path = "../crates/test-utils" }
 json-utils = { path = "../crates/json-utils" }
-fluence-app-service = "0.1.30"
+fluence-app-service = { git = "https://github.com/fluencelabs/fce", branch = "file_name" }
 fluence-libp2p = { path = "../crates/libp2p" }
 base64 = "0.13.0"
 maplit = "1.0.2"
 serde_json = "1.0.61"
+blake3 = "0.3.7"
```
Comment (michaelvoronov) [inline particle-node/Cargo.toml:61]: newline

File: particle-services/Cargo.toml [hunk 5]
```diff
@@ -22,3 +22,4 @@ serde = "1.0.116"
 log = "0.4.11"
 uuid = "0.8.1"
 toml = "0.5.6"
+thiserror = "1.0.23"
```
Comment (michaelvoronov) [inline particle-services/Cargo.toml:25]: newline

**Review by michaelvoronov — APPROVED** [review]

File: particle-modules/src/modules.rs (hunk 3 above)
Comment (folex) [reply particle-modules/src/modules.rs:127]: It's a nice library for errors http://lib.rs/eyre

**Pull request merged**